src/Futhark/CodeGen/Backends/CSOpenCL/Boilerplate.hs

```diff
@@ -0,0 +1,303 @@
+module Futhark.CodeGen.Backends.CSOpenCL.Boilerplate
+  ( generateBoilerplate
+
+  , kernelRuntime
+  , kernelRuns
+  ) where
+
+import qualified Data.Map as M
+
+import Futhark.CodeGen.ImpCode.OpenCL hiding (Index, If)
+import Futhark.CodeGen.Backends.GenericCSharp as CS
+import Futhark.CodeGen.Backends.GenericCSharp.AST as AST
+import Futhark.CodeGen.OpenCL.Kernels
+
+
+intT, longT, stringT, intArrayT, stringArrayT :: CSType
+intT = Primitive $ CSInt Int32T
+longT = Primitive $ CSInt Int64T
+stringT = Primitive StringT
+intArrayT = Composite $ ArrayT intT
+stringArrayT = Composite $ ArrayT stringT
+
+generateBoilerplate :: String -> String -> [String] -> [PrimType]
+                    -> M.Map VName (SizeClass, Name)
+                    -> CS.CompilerM OpenCL () ()
+generateBoilerplate opencl_code opencl_prelude kernel_names types sizes = do
```

Comment:
I think it is better if all publicly visible API names follow the usual C# [naming conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions).  It will make the generated code feel more like a real .NET library.

Reply:
What are you thinking of in particular?